File: strategy_java/src/main/java/com/kmmania/strategy_training/SprintTraining.java
package com.kmmania.strategy_training;

public class SprintTraining implements AthleticTraining {
    private final TrainingSession trainingSession;

    public SprintTraining(TrainingSession trainingSession) {
        this.trainingSession = trainingSession;
    }

    @Override
    public void train() {
        System.out.println("*** Sprint training ***");
        System.out.println("Date: " + this.trainingSession.date());
        System.out.println("Time slot: " + this.trainingSession.timeSlot());
        System.out.println("Place: " + this.trainingSession.place());
    }
}

File: strategy_kotlin/src/main/kotlin/strategy_training/SprintTraining.kt
/*
 * Strategy project.
 * Dac Linh Giao NGUYEN
 * License MIT
 *
 * Strategy project in Kotlin.
 * SprintTraining.kt
 */
package com.kmmania.strategy_training

/**
 * Implementation of the [AthleticTraining] interface representing sprint training strategy.
 * This class allows training planification for sprint sessions.
 *
 * @author Dac Linh Giao NGUYEN
 * @property trainingSession The training session details (date, time slot, place).
 * @constructor Constructs a new instance of SprintTraining with the provided TrainingSession.
 * @param trainingSession The training session details (date, time slot, place).
 * @since 0.1
 * @see AthleticTraining
 */
class SprintTraining(
    private val trainingSession: TrainingSession
) : AthleticTraining {

    /**
     * Method to perform the sprint training strategy.
     * This method prints details of the training session, including date, time slot, and place.
     */
    override fun train() {
        println("*** Sprint training ***")
        println("Date: ${trainingSession.date}")
        println("Time slot: ${trainingSession.timeSlot}")
        println("Place: ${trainingSession.place}")
    }
}